src/eval.rs

```diff
@@ -201,7 +201,8 @@ pub fn eval_main<'tcx>(tcx: TyCtxt<'tcx>, main_id: DefId, config: MiriConfig) ->
     // Perform the main execution.
     let res: InterpResult<'_, i64> = (|| {
         // Main loop.
-        while ecx.step()? {
+        while ecx.schedule()? {
+            assert!(ecx.step()?);
```

Comment:
Please add an assert message here, to make it easier to see what is being asserted.